pmd-core/src/main/java/net/sourceforge/pmd/lang/document/Chars.java

```diff
@@ -520,69 +520,74 @@ boolean isFullString() {
      * a line delimiter is {@code CR}, {@code LF} or {@code CR+LF}.
      */
     public Iterable<Chars> lines() {
-        return () -> new Iterator<Chars>() {
-            final int max = len;
-            int pos = 0;
-            // If those are NOT_TRIED, then we should scan ahead to find them
-            // If the scan fails then they'll stay -1 forever and won't be tried again.
-            // This is important to scan in documents where we know there are no
-            // CR characters, as in our normalized TextFileContent.
-            int nextCr = NOT_TRIED;
-            int nextLf = NOT_TRIED;
+    return () -> new LineIterator();
+}
 
-            @Override
-            public boolean hasNext() {
-                return pos < max;
-            }
+private class LineIterator implements Iterator<Chars> {
+    final int max = len;
+    int pos = 0;
+    int nextCr = NOT_TRIED;
+    int nextLf = NOT_TRIED;
 
-            @Override
-            public Chars next() {
```

Comment:
I don't think these are positive changes. Moving this to an inner class instead of an anonymous one can be defended, but you removed all comments, and splitting the implementation into separate methods makes it really harder to follow IMO.